### PR TITLE
Make treasure chests diggable and drop their contents when dug

### PR DESCRIPTION
**Pull request by realOneplustwo**

### CODE REVIEW

**Comment by LoneWolfHT** [comment]
This doesn't make treasure chests diggable anymore

File: mods/other/tsm_chests/init.lua [hunk 1]
```diff
@@ -58,10 +58,21 @@ minetest.register_node("tsm_chests:chest", {
...
 		inv:set_size("main", 8*4)
 	end,
-	can_dig = function(pos,player)
+	drop = "",
+	on_destruct = function(pos)
 		local meta = minetest.get_meta(pos);
 		local inv = meta:get_inventory()
-		return inv:is_empty("main")
+		-- iterate over chest contents, drop all
+		for i = 1, inv:get_size("main") do
```
Comment (LoneWolfHT) [inline mods/other/tsm_chests/init.lua:66]: Pretty sure you can just do
```lua
for k, itemstack in pairs(inv:get_list("main") do```

File: mods/other/tsm_chests/init.lua [hunk 2]
```diff
@@ -58,10 +58,21 @@ minetest.register_node("tsm_chests:chest", {
...
+			local stack = inv:get_stack("main", i)
+			if not minetest.is_yes(minetest.settings:get("remove_items")) then
+				local obj = minetest.add_item(pos, stack:take_item(stack:get_count()))
+				if obj then
+					obj:set_velocity({x = math.random(-1, 1), y = 5, z = math.random(-1, 1)})
+				end
+			end
+			inv:set_stack("main", i, nil)
```
Comment (LoneWolfHT) [inline mods/other/tsm_chests/init.lua:74]:
```suggestion
			inv:set_stack("main", i)
```

nil is the value given when an argument isn't passed

**Review by LoneWolfHT — CHANGES_REQUESTED** [review]
nvm about my last comment, Github being dumb

File: mods/other/tsm_chests/init.lua [hunk 3]
```diff
@@ -58,10 +59,17 @@ minetest.register_node("tsm_chests:chest", {
...
-	can_dig = function(pos,player)
+	on_destruct = function(pos)
 		local meta = minetest.get_meta(pos);
 		local inv = meta:get_inventory()
-		return inv:is_empty("main")
+		-- iterate over chest contents, drop all
+		for index, stack in pairs(inv:get_list("main")) do
+			if not minetest.is_yes(minetest.settings:get("remove_items")) then
+				local obj = minetest.add_item(pos, stack:take_item(stack:get_count()))
+				_ = obj and obj:set_velocity({x = math.random(-1, 1), y = 5, z = math.random(-1, 1)})
```
Comment (Panquesito7) [inline mods/other/tsm_chests/init.lua:69]: Is this variable even being used? It should be marked as `local` (see [LuaCheck](https://github.com/MT-CTF/capturetheflag/pull/748/checks?check_run_id=2042105702) checks).
```suggestion
				local _ = obj and obj:set_velocity({x = math.random(-1, 1), y = 5, z = math.random(-1, 1)})
```

Comment (realOneplustwo) [reply mods/other/tsm_chests/init.lua:69]: I thought underscore was a throwaway variable.

Comment (realOneplustwo) [reply mods/other/tsm_chests/init.lua:69]: Anyway, the purpose of line 69 is just to call obj:set_velocity after checking obj is not nil.

**Pull request closed without merging**